Sources/OpenAPIRuntime/EventStreams/ServerSentEventsDecoding.swift

```diff
@@ -79,26 +100,34 @@ extension AsyncSequence where Element == ArraySlice<UInt8>, Self: Sendable {
     /// Returns another sequence that decodes each event's data as the provided type using the provided decoder.
     ///
     /// Use this method if the event's `data` field is not JSON, or if you don't want to parse it using `asDecodedServerSentEventsWithJSONData`.
+    /// - Parameter: An optional closure that determines whether the given byte sequence is the terminating byte sequence defined by the API.
     /// - Returns: A sequence that provides the events.
-    public func asDecodedServerSentEvents() -> ServerSentEventsDeserializationSequence<
+    public func asDecodedServerSentEvents(terminate: (@Sendable (ArraySlice<UInt8>) -> Bool)? = nil) -> ServerSentEventsDeserializationSequence<
         ServerSentEventsLineDeserializationSequence<Self>
-    > { .init(upstream: ServerSentEventsLineDeserializationSequence(upstream: self)) }
+    > { .init(upstream: ServerSentEventsLineDeserializationSequence(upstream: self), terminate: terminate) }
```

Comment:
```suggestion
    > { .init(upstream: ServerSentEventsLineDeserializationSequence(upstream: self).prefix(while: { !terminate($0) }) }
```

Reply:
I'm getting 

> Cannot convert value of type 'AsyncPrefixWhileSequence<ServerSentEventsLineDeserializationSequence<Self>>' to expected argument type 'ServerSentEventsLineDeserializationSequence<Self>'

with

```swift
> { .init(upstream: ServerSentEventsLineDeserializationSequence(upstream: self).prefix(while: { if let terminate { return terminate($0) } else { return false } })) }
```

Should I be making the `prefix` call on the `upstream` paramter instead? Although that doesn't appear to work either.

Reply:
Oh of course, I forgot we're not returning `some AsyncSequence<...>`, hmm.

I wonder if we should only add this for Swift 6 compiler and higher with `#if compiler(>=6)` (which doesn't mean we need Swift 6 language mode, however).

Otherwise we'll really need to pass this parameter into `ServerSentEventsLineDeserializationSequence` and make it implement the stopping (which is also not the worst outcome).

Thoughts, @simonjbeaumont?

Reply:
My hunch is to _not_ make this conditional on a newer compiler because we'll just push the burden to adopters. It's fine when the adopter is a leaf package, but if they want to provide a library that supports older compilers they'll have to do a conditional-compilation dance too.

Reply:
Yeah agreed - let just sink down the closure where it needs to be.

Reply:
> Yeah agreed - let just sink down the closure where it needs to be.

I.e. keep it the way it is in the PR now?